src/coprocessor/endpoint.rs

```diff
@@ -1049,12 +1025,15 @@ mod tests {
         const PAYLOAD_LARGE: i64 = 6000;
 
         let pd_worker = FutureWorker::new("test-pd-worker");
-        let engine = TestEngineBuilder::new().build().unwrap();
-        let read_pool = ReadPool::new(
-            "readpool",
+
+        let read_pool = ReadPoolImpl::build_read_pool(
```

Comment:
For this one maybe we can use `Builder::from_config(..).build()` (because we don't need on_tick or before_stop in this test). Similar for others.